lang/cs/Org.Apache.REEF.Tests/Functional/FaultTolerant/TestResubmitContext.cs

```diff
@@ -0,0 +1,235 @@
+
+            private readonly IEvaluatorRequestor _requestor;
+            private int _taskNumber = 1;
+            private int _contextNumber = 1;
+
+            [Inject]
+            private ResubmitEvaluatorDriver(IEvaluatorRequestor requestor)
+            {
+                _requestor = requestor;
+            }
+
+            public void OnNext(IDriverStarted value)
+            {
+                _requestor.Submit(_requestor.NewBuilder().SetNumber(2).Build());
+            }
+
+            public void OnNext(IAllocatedEvaluator value)
+            {
+                Logger.Log(Level.Info, "AllocatedEvaluator: " + value.Id);
+                value.SubmitContext(
+                    ContextConfiguration.ConfigurationModule
+                        .Set(ContextConfiguration.Identifier, "ContextID" + _contextNumber++)
+                        .Build());
+            }
+
+            public void OnNext(IActiveContext value)
+            {
+                Logger.Log(Level.Info, "ActiveContext: " + value.Id);
+                value.SubmitTask(TaskConfiguration.ConfigurationModule
+                    .Set(TaskConfiguration.Identifier, TaskId + _taskNumber++)
+                    .Set(TaskConfiguration.Task, GenericType<FailEvaluatorTask>.Class)
+                    .Set(TaskConfiguration.OnMessage, GenericType<FailEvaluatorTask>.Class)
+                    .Set(TaskConfiguration.OnClose, GenericType<FailEvaluatorTask>.Class)
+                    .Build());
+            }
+
+            public void OnNext(IRunningTask value)
+            {
+                Logger.Log(Level.Info, "RunningTask: " + value.Id);
+                string msg = value.Id.Equals(TaskId + "1") ? FailSignal : SuccSignal;
+                value.Send(Encoding.UTF8.GetBytes(msg));
+            }
+
+            public void OnNext(ICompletedTask value)
+            {
+                Logger.Log(Level.Info, CompletedTaskValidationMessage + ". Task completed: " + value.Id);
+                value.ActiveContext.Dispose();
+            }
+
+            public void OnNext(ICompletedEvaluator value)
+            {
+                Logger.Log(Level.Info, CompletedEvaluatorValidationMessage + ". Evaluator completed: " + value.Id);
+            }
+
+            public void OnNext(IFailedEvaluator value)
+            {
+                Logger.Log(Level.Info, FailedEvaluatorMessage + ". Evaluator failed: " + value.Id + "  FailedTaskId: " + value.FailedTask.Value.Id);
+                foreach (var c in value.FailedContexts)
+                {
+                    Logger.Log(Level.Info, "Failed Context: " + c.Id);
+                    Assert.Equal(c.Id, "ContextID1");
+                }
+                _requestor.Submit(_requestor.NewBuilder().Build());
+            }
+
+            public void OnNext(IFailedContext value)
+            {
+                Logger.Log(Level.Info, "In IFailedContext handler: " + value.Id);
```

Comment:
I believe the failure is intentional -- the test is to test the resubmission of an `Evaluator`.